### PR TITLE
support PriorityClass on backend

### PR DESCRIPTION
**Pull request by khalilswdp**
Added Service Layer Interface, Service, Handler, DI to container, as well as the handler and paths to the server.
Can I have some guidance on how to test and possible next steps? Thank you

### CODE REVIEW

**Comment by khalilswdp** [comment]
Just signed the CLA

**Comment by sanposhiho** [comment]
Hi @khalilswdp. Thanks for creating the PR. I'll review it later.

/retitle support PriorityClass on backend
/kind feature
/label tide/merge-method-squash

**Comment by sanposhiho** [comment]
@khalilswdp 

> Can I have some guidance on how to test and possible next steps? Thank you

For now, we don't have unit tests for most of packages on service layer and handler. So, you don't need to add tests.
After this PR gets merged, we have to support PriorityClass on frontend(web UI) for the next step of #27. 
Do you want to develop web UI as well? If you want, I can give you some help. If you don't want, I can develop frontend for #27.

The development of the frontend requires the skills/knowledge of Vue3, Nuxt, and TypeScript.

**Comment by khalilswdp** [comment]
@sanposhiho  I want to develop the web UI as well! I think that'll help me learn a bit more how things fit together. But I still need help

**Comment by khalilswdp** [comment]
Hi @sanposhiho , I have been trying to do some work on the front end, but I am facing problems with `yarn dev` telling me Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource. But this doesn't happen when I do `yarn build` followed by `yarn start`, or when I do make `make docker_build_and_up`. 
Do you know what's the problem or what can I do in order to avoid CQRS when doing development?

**Comment by khalilswdp** [comment]
In the frontend, the button doesn't bring up a resource template, I don't know why

**Comment by khalilswdp** [comment]
Can I squash commits together? Or use `git reset --hard HEAD` to reorganize commits ? Or is it fine this way?

**Comment by sanposhiho** [comment]
@khalilswdp 
Could you split this PR into two PRs. (one is for API change and another is for frontend change)

**Comment by sanposhiho** [comment]
>  I have been trying to do some work on the front end, but I am facing problems with yarn dev telling me Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource. But this doesn't happen when I do yarn build followed by yarn start, or when I do make make docker_build_and_up.
Do you know what's the problem or what can I do in order to avoid CQRS when doing development?

Backend specify config.FrontendURL as `AllowOrigins`.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/server/server.go#L30

So, I think it should be resolved by passing your frontend URL via environment variable `FRONTEND_URL `.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/config/config.go#L69

**Comment by khalilswdp** [comment]
I separated frontend and backend changes as requested

File: priorityclass/priorityclass.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package priorityclass
+
+import (
+	"context"
```
Comment (sanposhiho) [inline priorityclass/priorityclass.go:4]: Please sort your import with `make lint-fix`.

File: priorityclass/priorityclass.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+// NewPriorityClassService initializes Service.
+func NewPriorityClassService(client clientset.Interface) *Service {
+	return &Service{
+		client: client,
+	}
+}
+
+func (s Service) Get(ctx context.Context, name string) (*v1.PriorityClass, error) {
```
Comment (sanposhiho) [inline priorityclass/priorityclass.go:24]: Please fix all receiver to pointer like other service.

```suggestion
func (s *Service) Get(ctx context.Context, name string) (*v1.PriorityClass, error) {
```

File: server/di/di.go [hunk 3]
```diff
@@ -67,6 +70,12 @@ func (c *Container) PersistentVolumeClaimService() PersistentVolumeClaimService
 	return c.pvcService
 }
 
+// SchedulerService returns SchedulerService.
```
Comment (sanposhiho) [inline server/di/di.go:73]: Thanks :)

**Review by sanposhiho — CHANGES_REQUESTED** [review]
Hi @khalilswdp. Nice contribution.
I left some comments.

**Comment by sanposhiho** [comment]
@khalilswdp 
And, could you please add new API information to this api.md.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/docs/api.md

File: priorityclass/priorityclass.go (hunk 1 above)
Comment (khalilswdp) [reply priorityclass/priorityclass.go:4]: I've issued the command on the root of the project

Comment (khalilswdp) [reply priorityclass/priorityclass.go:4]: But it doesn't seem like it's doing much

File: priorityclass/priorityclass.go (hunk 2 above)
Comment (khalilswdp) [reply priorityclass/priorityclass.go]: My mistake, I've changed the receiver from struct to struct pointer

File: priorityclass/priorityclass.go (hunk 1 above)
Comment (sanposhiho) [reply priorityclass/priorityclass.go:4]: It's ok. All imports are sorted correctly.

**Comment by khalilswdp** [comment]
I have just added the api calls to api.md

**Review by sanposhiho — COMMENTED** [review]
Looks good.
Thanks for the contribution!

/lgtm
/approve

**Comment by sanposhiho** [comment]
For information: This PR is related to #27

**Pull request merged**